lib/closex/caching_client.ex

```diff
@@ -14,14 +14,26 @@ defmodule Closex.CachingClient do
   end
 
   defdelegate create_lead(payload, opts \\ []), to: @fallback_client
-  defdelegate update_lead(lead_id, payload, opts \\ []), to: @fallback_client
+
+  def update_lead(lead_id, payload, opts \\ []) do
+    case apply(@fallback_client, :update_lead, [lead_id, payload, opts]) do
+      {:ok, updated_lead} -> {:ok, set_cached(lead_id, updated_lead)}
```

Comment:
Prefer to set the return value on a newline like so:

```
{:ok, updated_lead} ->
  {:ok, set_cached(lead_id, updated_lead)}
```

Reply:
You can even use `with` here for brevity.

```elixir
with {:ok, updated_lead} <- apply(@fallback_client, :update_lead, [lead_id, payload, opts]) do
  {:ok, set_cached(lead_id, updated_lead)}
end
```

Reply:
Sure thing :+1:. Out of interest, why the preference for:

```elixir
{:ok, updated_lead} ->
  {:ok, set_cached(lead_id, updated_lead)}
{:error, reason} ->
  {:error, reason}
```
over
```elixir
{:ok, updated_lead} -> {:ok, set_cached(lead_id, updated_lead)}
{:error, reason} -> {:error, reason}
```
I personally find the latter easier to read because it makes the different cases on the left hand side more clear.  Happy to change of course! Just curious :)